e2e/playwright/info-drawer/src/tests/file-folder-properties.e2e.ts

```diff
@@ -23,13 +23,60 @@
  */
 
 import { expect } from '@playwright/test';
-import { ApiClientFactory, Utils, test, TrashcanApi, NodesApi, FileActionsApi } from '@alfresco/aca-playwright-shared';
+import {
+  ApiClientFactory,
+  Utils,
+  test,
+  TrashcanApi,
+  NodesApi,
+  FileActionsApi,
+  TagsApi,
+  CategoriesApi,
+  PersonalFilesPage
+} from '@alfresco/aca-playwright-shared';
 
 test.describe('Info Drawer - File Folder Properties', () => {
   let nodesApi: NodesApi;
   let trashcanApi: TrashcanApi;
   let fileActionsApi: FileActionsApi;
+  let tagsApi: TagsApi;
+  let categoriesApi: CategoriesApi;
+  let responseCategoryId: string;
+  let responseTagsId: string;
   const username = `user-e2e-${Utils.random()}`;
+  const manualTagName = `e2e-tag-${Utils.random()}`;
+  const FolderC299162 = `C299162-e2e-${Utils.random()}`;
+  const FolderC599174 = `C599174-e2e-${Utils.random()}`;
+  const Folder17238 = `xat-17238-e2e-${Utils.random()}`;
+  const Folder17239 = `xat-17239-e2e-${Utils.random()}`;
+  const Folder17240 = `xat-17240-e2e-${Utils.random()}`;
+  const Folder17241 = `xat-17241-e2e-${Utils.random()}`;
+  const Folder17242 = `xat-17242-e2e-${Utils.random()}`;
+  const Folder17243 = `xat-17243-e2e-${Utils.random()}`;
+  const Folder17244 = `xat-17244-e2e-${Utils.random()}`;
+  const tagBody = { tag: `tag-${Utils.random()}` };
+  const categoryId2 = `-root-`;
+  const categoryName = Utils.random();
+
+  async function createCategoryGetId() {
+    const requestData = await categoriesApi.createCategory(categoryId2, [{ name: categoryName }]);
+    if ('entry' in requestData) {
+      return requestData.entry.id;
+    } else {
+      console.error('Unexpected response format:', requestData);
+      return null;
+    }
+  }
+
+  async function createTagGetId() {
+    const requestTagData = await tagsApi.createTags([tagBody]);
+    if (Object.keys(requestTagData).includes('entry')) {
```

Comment:
we can replace this if with better code like below to replace unknown type checking.
if ('entry' in requestTagData) {
            return (requestTagData as { entry: { id: string } }).entry.id;
        }

Reply:
I was using 'entry' in request but sonar cloud is giving me this error:
`**_Use "indexOf" or "includes" (available from ES2016) instead._**`

I switched it back to using 'in' keyword and the issue occurs again. What should I do here?

Reply:
https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=4210&id=Alfresco_alfresco-content-app&open=AZLhr2TdcCVkb1Q-mrj7

Reply:
@AleksanderSklorz the sonar cloud issue is discussed here.

Reply:
Hmm I see. Honestly not sure why Sonar is complaining about that. Using 'in' should be fine as well in my opinion  as it looks simpler. @MichalKinas is there any reason why we can't use 'in'? Should we somehow change it for sonar?

Reply:
In this case I think Sonar is right since `requestTagData` is an array an `in` keyword is TS work for objects, in case of arrays it won't check values but rather indexes of array so this could lead to some errors, here since you're creating only one tag you can use `'entry' in requestTagData[0]` and Sonar should be okay with that

Reply:
ah, good catch :D I didn't notice it's array. Variable is named requestTagData so I thought it's some object with tag data :D

Reply:
Also this line `return (requestTagData as { entry: { id: string } }).entry.id;` looks a bit weird, since you know you created only one tag you can just return `return requestTagData[0].entry.id;`

Reply:
@MichalKinas With my code the tests pass normally, but with your suggestions I get this error:

beforeAll failed : TypeError: Cannot use 'in' operator to search for 'entry' in undefined

Reply:
Can you please put a breakpoint there or console log what api returns? I just want to make sure what is the type of returned object

Reply:
Weirdly, it creates 3 tags.
const createdTag = await tagsApi.createTags([tagBody]);

returns this:
{
  entry: { tag: 'tag-1jj01', id: '52d24d73-48b3-418a-924d-7348b3e18a2f' }
}
{
  entry: { tag: 'tag-19453', id: '963abe54-1e21-4e78-babe-541e21ce7877' }
}
{
  entry: { tag: 'tag-13d1t', id: '9db166f6-8c5c-4ac8-b166-f68c5c7ac8eb' }
}

and all of them are deleted after the test.

Reply:
Yeah so as we were speaking it returns array of tags (don't know why 3 though) so you can't use `if ('entry' in createdTag)` but rather `if ('entry' in createdTags[0])`

Reply:
The tests just doesn't work when I try to use `if ('entry' in createdTags[0])`

`  async function createTagGetId(): Promise<string> {
    const createdTag = await tagsApi.createTags([tagBody]);
    if ('entry' in createdTag[0]) {
      return createdTag[0].entry.id;
    } else {
      console.error('Unexpected response format:', createdTag);
      return null;
    }
  }`

I get this error:
beforeAll failed : TypeError: Cannot use 'in' operator to search for 'entry' in undefined

Then the tests all fail.

Reply:
The tests just doesn't work when I try to use `if ('entry' in createdTags[0])`

`  async function createTagGetId(): Promise<string> {
    const createdTag = await tagsApi.createTags([tagBody]);
    if ('entry' in createdTag[0]) {
      return createdTag[0].entry.id;
    } else {
      console.error('Unexpected response format:', createdTag);
      return null;
    }
  }`

I get this error:
beforeAll failed : TypeError: Cannot use 'in' operator to search for 'entry' in undefined

Then the tests all fail.

Reply:
and if I try this:

  async function createTagGetId(): Promise<string> {
    const createdTag = await tagsApi.createTags([tagBody]);
    if ('entry' in createdTag) {
      return createdTag[0].entry.id;
    } else {
      console.error('Unexpected response format:', createdTag);
      return null;
    }
  }

I get this error:
beforeAll failed : TypeError: Cannot read properties of undefined (reading 'entry')

I guess the TagEntry type is the problem here, but I don't understand why. @MichalKinas

Reply:
That's why I used this line of code `return (createdTag as { entry: { id: string } }).entry.id;`, because it treats the createdTag as a specific object with entry and id, and not as TagEntry which is somehow interpreted differently in this case.

Reply:
So in your first case you got `beforeAll failed : TypeError: Cannot use 'in' operator to search for 'entry' in undefined` which means that in that case `createdTag[0]` is undefined which means that `tagsApi.createTags` didn't return tags, the rest of the code should work fine I even tested that in jsconsole -> 
<img width="324" alt="Zrzut ekranu 2024-11-5 o 09 45 43" src="https://github.com/user-attachments/assets/7529cd74-a7c7-40b0-b6a4-192d37182ae7">

Reply:
If you still get the error please let me know how to run this case locally I will try to debug

Reply:
@MichalKinas in both cases the tags are created. This is the case with createdTag[0]. I still don't understand why it creates 3 tags, but I don't think this is the issue right now. The code is up-to-date on this branch, so if you want to debug it locally, you can do it. You can message me if you encounter any issues with running the e2es.

createdTag: {
  entry: { tag: 'tag-1ld4r', id: '222ce83c-e6a1-487c-ace8-3ce6a1487c98' }
}
beforeAll failed : TypeError: Cannot use 'in' operator to search for 'entry' in undefined
createdTag: {
  entry: { tag: 'tag-10mri', id: '2c78ac98-e2a9-44c8-b8ac-98e2a9a4c894' }
}
beforeAll failed : TypeError: Cannot use 'in' operator to search for 'entry' in undefined
createdTag: {
  entry: { tag: 'tag-137wp', id: 'bcf4dce5-93be-4955-b4dc-e593be395537' }
}
beforeAll failed : TypeError: Cannot use 'in' operator to search for 'entry' in undefined